libtransmission/rpc-server.c

```diff
@@ -685,7 +685,8 @@ static void handle_request(struct evhttp_request* req, void* arg)
 
         server->loginattempts = 0;
 
-        if (strncmp(req->uri, server->url, strlen(server->url)) != 0)
+        if (strstr(req->uri, server->url) == NULL || req->uri[strlen(server->url)] == '\0' ||
+            (strncmp(req->uri + strlen(server->url), "web", 3) == 0 && req->uri[strlen(server->url) + 3] == '\0'))
```

Comment:
What you're really trying to do here I guess is to move the `location` declaration out of the `if` statement (remembering to properly free it, and maybe renaming it in the process) and then do `if (strncmp(req->uri, location, strlen(location)) != 0)`. What you currently have is a bit complex and inefficient given that `strlen(server->url)` is being calculated three times in the worst case despite always returning the same value.

Reply:
Thank you, I refactored this code and made it simpler.

I've already thought about multiple strlens, but decided that it could be optimized out by modern compilers because there are no string mutations between them and all functions there work with const char*.
Anyway, this is not actual now.

Reply:
`const` in C/C++ means "not mutable through this particular variable", not "completely immutable". Optimizing out `strlen` calls would be irresponsible.